llvm/include/llvm/IR/RepoDefinition.h

```diff
@@ -49,17 +49,60 @@ void set(GlobalObject *GO, DigestType const &D);
 /// GO does not contain the RepoDefinition metadata.
 std::pair<DigestType, bool> get(const GlobalObject *GO);
 
-/// A structure of a global object (GO) information.
+/// A structure of a global object (GO) information, which is used to calculate
+/// the GO's digest. It includes the GO's initial degist, initial/final
+/// 'Dependencies', and its `Contributions` information.
 struct GOInfo {
   /// The InitialDigest is the hash of an object itself (a function or global
   /// variable) without consideration for any references to other objects.
   DigestType InitialDigest;
+
   /// The Contributions of an object (X) are those objects (Y) which
   /// transitively reference X and where a potential optimisation to either X or
   /// any of Y may invalidate both.
   GOVec Contributions;
-  /// The Dependencies of an object are those objects which it transitively
-  /// references but are not Contributions.
+
+  /// When we calculate the initial digest for each global object (GO), we also
+  /// collect the GO's 'Contributions' and 'Dependencies'. The GO's
+  /// 'Contributions' and 'Dependencies' represent the different directions
+  /// during the hash calculation. If GO's digest might be affected by other GOs
+  /// and GO also might modify other GOs, we will put other GOs into the GO's
+  /// 'Contributions'. However, the 'Dependencies' of the GO are those objects
+  /// whose digest value changes might affect the GO's digest value.
+  ///
+  /// Consider the following example:
+  /// static int G = 0;
+  ///  void setTo(int* const P, int V) {
+  ///      *P += V;
```

Comment:
Yes, I checked using Clang-format. LLVM uses 2 spaces, but pstore project uses 4 spaces.

Reply:
>> Use 2 (rather than 4) space indents to fit with the LLVM style?
> LLVM uses 2 spaces, but pstore project uses 4 spaces.

True, although I’m not sure that the pstore convention should be significant here: this is LLVM code.

Reply:
Sorry, missed several lines. Fixed in commit becfdbf3482387.